l10n_it_intrastat_statement/models/intrastat_statement.py

```diff
@@ -775,35 +775,42 @@ def compute_statement(self):
             inv_type += ["in_invoice", "in_refund"]
         domain.append(("move_type", "in", inv_type))
 
-        statement_data = dict()
+        statement_data = {}
+        section_field_reverse = {}
+        for section_type in ["purchase", "sale"]:
+            for section_number in range(1, 5):
+                statement_section_field = self.get_section_field_name(
+                    section_type, section_number
+                )
+                statement_data[statement_section_field] = list()
```

Comment:
```suggestion
        statement_data = defaultdict(list)
        section_field_reverse = {}
        for section_type in ["purchase", "sale"]:
            for section_number in range(1, 5):
                statement_section_field = self.get_section_field_name(
                    section_type, section_number
                )
```
importando prima `from collections import defaultdict`.
In questo modo torna il comportamento precedente (di cui abbiamo già parlato in https://github.com/OCA/l10n-italy/pull/3888/files#r1466248365) di valorizzare solo i campi che avranno delle righe, ed evitiamo di creare delle liste vuote e doverle poi rimuovere più sotto nella _dictionary comprehension_ (quindi la `write` potrà tornare com'era prima).
Cosa ne pensi?

Reply:
Ok! Lo cambio